generate/generate.py

```diff
@@ -246,12 +248,20 @@ def generatePath(
             else:
                 print("  parameter: ", parameter)
                 raise Exception("Unknown parameter type")
-            f.write(
-                ", " +
-                parameter_name +
-                "=" +
-                camel_to_snake(parameter_name))
-    f.write(")\n")
+            if parameter['in'] == 'path':
+                formatTemplate = formatTemplate + ", " + \
+                    parameter_name + "=" + camel_to_snake(parameter_name)
+            elif parameter['in'] == 'query':
+                formatTemplate = formatTemplate + ", " + \
+                    parameter_name + "=" + camel_to_snake(parameter_name)
+                queryTemplate = queryTemplate + "&" + \
+                    camel_to_snake(parameter_name) + "=" + "{" + camel_to_snake(parameter_name) + "}"
+    if queryTemplate[1:].__len__() > 0:
+        f.write("\turl = \"" + templateUrl + "?" +
+                queryTemplate[1:] + "\"" + formatTemplate + ")\n")
+    else:
+        f.write("\turl = \"" + templateUrl + "\"" + formatTemplate + ")\n")
+
```

Comment:
instead of `f.write`ing in the loop, it mutates some strings that are than assembled after the loop because it needs to change the template and the format params.

If there's a better way to do this, do let me know.